[ESC] Update OIDC Documentation for vault-login

ejhayes opened the pull request:
## Description
Fixes: https://github.com/pulumi/docs/issues/10193

This updates the ESC documentation for `vault-login` to show how to setup Vault OIDC authentication to Pulumi Cloud.

## Checklist:

- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [x] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [x] I have manually confirmed that all new links work.
- [x] I added aliases (i.e., redirects) for all filename changes.
- [x] If making css changes, I rebuilt the bundle.

toriancrane commented on themes/default/content/docs/pulumi-cloud/esc/providers/vault-login.md:
````diff
@@ -24,6 +24,62 @@ The `vault-login` provider enables you to log in to HashiCorp Vault using OpenID
+```
+
+### Create JWT Role
+To create a [role](https://developer.hashicorp.com/vault/api-docs/auth/jwt#create-update-role) (assumes existing policy `myExamplePolicy`) using the CLI:
````
```suggestion
To create a [role](https://developer.hashicorp.com/vault/api-docs/auth/jwt#create-update-role) (assumes existing policy `myExamplePolicy`) using the CLI:
```

Would you mind including the steps to create a policy? It can be a simple one with a note for the user to make sure the policy includes the permissions that they need for their own usage.

ejhayes replied:
Good idea, I've moved this @toriancrane

toriancrane reviewed (changes requested):
Hey @ejhayes, thank you for writing this! Would it be possible to migrate this content to it's own page as a child of the `OpenID Connect` page? (i.e. to live on a page titled `Configuring OpenID with Vault` and having its page hierarchy be `/docs/pulumi-cloud-oidc/vault`). We'd like to keep the OIDC content consolidated to make it easier for users to find/use.

It would be great if you could also incorporate the 'Pulumi-side' steps i.e. how to configure a Pulumi ESC environment to use this role and how to configure Pulumi Deployments to do the same.

Feel free to use the [OIDC for AWS gudie](https://www.pulumi.com/docs/pulumi-cloud/oidc/aws/) as a reference.

ejhayes commented:
@toriancrane I've addressed your feedback in the latest commit. Hopefully this will be sufficient.

Note that I also opened up pulumi/pulumi-cloud-requests/issues/322 to track adding deployment specific vault oidc configuration. Not sure if this is even needed with ESC anymore but wanted to open so it could be referenced in the docs.

toriancrane commented on themes/default/content/docs/pulumi-cloud/oidc/vault.md:
```diff
@@ -0,0 +1,192 @@
+- /docs/pulumi-cloud/deployments/oidc/vault/
+---
+
+This document outlines the steps required configured to use Pulumi to use OpenID Connect to authenticate with Vault. This is accomplished using [Vault's JWT authentication method](https://developer.hashicorp.com/vault/docs/auth/jwt#jwt-authentication) to assume a role. Access to the role is authorized using a [Vault policy](https://developer.hashicorp.com/vault/docs/concepts/policies) that valida ...
```
```suggestion
This document outlines the steps required configured to use Pulumi to use OpenID Connect to authenticate with Vault. This is accomplished using [Vault's JWT authentication method](https://developer.hashicorp.com/vault/docs/auth/jwt#jwt-authentication) to assume a role. Access to the role is authorized using a [Vault policy](https://developer.hashicorp.com/vault/docs/concepts/policies) that validates the contents of the OIDC token issued by Pulumi Cloud.

{{% notes "info" %}}
The `namespaces` functionality of Vault is not currently supported. More specifically, this configuration will only work for the `root` namespace. This means that this configuration will only work with Vault OSS at this time (Vault HCP and Enterprise are not supported).
{{% /notes %}}
```

toriancrane commented on themes/default/content/docs/pulumi-cloud/esc/providers/vault-login.md:
```diff
@@ -13,6 +13,8 @@ menu:
 
 The `vault-login` provider enables you to log in to HashiCorp Vault using OpenID Connect or by providing static credentials. The provider will return a set of credentials that can be used to fetch secrets using the `vault-secrets` provider.
```
```suggestion
The `vault-login` provider enables you to log in to HashiCorp Vault using OpenID Connect or by providing static credentials. The provider will return a set of credentials that can be used to fetch secrets using the `vault-secrets` provider.

{{% notes "info" %}}
The `namespaces` functionality of Vault is not currently supported. More specifically, this configuration will only work for the `root` namespace. This means that this configuration will only work with Vault OSS at this time (Vault HCP and Enterprise are not supported).
{{% /notes %}}
```

toriancrane reviewed (changes requested):
Hi @ejhayes,

Thanks a lot for those updates!

I've had some time to do a deeper dive on the content, and I've included some suggested edits in this review.

ejhayes commented on themes/default/content/docs/pulumi-cloud/oidc/vault.md:
```diff
@@ -0,0 +1,192 @@
+
+### Pulumi Deployments
+
+{{% notes "info" %}}
```
I like your note much better than what I had, I've removed this.

ejhayes commented on themes/default/content/docs/pulumi-cloud/oidc/vault.md:
```diff
@@ -0,0 +1,192 @@
+Deployment specific Vault OIDC functionality is not currently supported. A [request](https://github.com/pulumi/pulumi-cloud-requests/issues/322) has been opened requesting this functionality. The notes below refer to using ESC with Pulumi Deployments to fetch secrets.
+{{% /notes %}}
+
+You can pull vault secrets from Pulumi ESC in Deployments. To set this up:
```
Agreed, I've removed this.

ejhayes commented on themes/default/content/docs/pulumi-cloud/oidc/vault.md:
````diff
@@ -0,0 +1,192 @@
+    {{< video title="Creating a new Pulumi ESC environment" src="./aws/create-new-environment.mp4" autoplay="true" loop="true" >}}
+  5. You will be presented with a split-pane editor view. Delete the default placeholder content in the editor and replace it with the following code:
+
+      ```yaml
````
Good catch 😄

ejhayes commented:
Ok I've read through the docs with all your changes and feedback incorporated. Thanks again @toriancrane.

toriancrane reviewed (approved):
Looks good to me!
